### PR TITLE
docs(message-guidelines): add message guidelines

### PR DESCRIPTION
**Pull request by xmfcx**
Signed-off-by: M. Fatih Cırıt mfc@leodrive.ai

## Description

I've decided to put this here instead. This way other message repos can benefit from this too.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
Made use of https://squidfunk.github.io/mkdocs-material/reference/admonitions/ they look cute.

File: docs/contributing/coding-guidelines/.pages
Comment (kenji-miyake) [inline docs/contributing/coding-guidelines/.pages:5]: It seems missing `.md`.
```suggestion
  - Message guidelines: message-guidelines.md
```

Also, I think you can place this under `ROS nodes coding guidelines`. (You can change the section name.)
I'll leave it to you.

Comment (xmfcx) [reply docs/contributing/coding-guidelines/.pages]: Thanks, I've fixed it. Also moved it under ros-nodes folder, it makes sense.

Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/.pages]: Thank you! (I have to write other guidelines ASAP... :pleading_face: )

File: docs/contributing/coding-guidelines/ros-nodes/.pages
Comment (kenji-miyake) [inline docs/contributing/coding-guidelines/ros-nodes/.pages:8]: The title here can be omitted.

```suggestion
  - message-guidelines.md
```

Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/ros-nodes/.pages:8]: Because the page title is used.
image

Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/.pages]: Ok, done in https://github.com/autowarefoundation/autoware-documentation/pull/94/commits/56e8eab84a2a7584825cfa45769e0de4ee4ac343

**Comment by xmfcx** [comment]
Ah before merging I need to add this too https://github.com/autowarefoundation/autoware_msgs/pull/11#pullrequestreview-948054486

Sorry let me turn it to draft again  :pleading_face:

File: docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md
Comment (kenji-miyake) [inline docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:135]: [Nits] `if available` is a bit unclear to me.
Could you clarify this? Is it different from `for non-default types`?

Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:135]: It is for non-default types, yes. Use the recommended ones, not something made up. 

Example: for denoting `km/h`, use `kmph`, not `km_h`

Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:135]: If available in the table.

Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:135]: So how about writing like "if available on the table of `Non-default units`", or rephrase it?

Also, if you use `if available`, I feel we should write the case of `if not available`. I guess it would "Please confirm the maintainers".

What do you think about it?

Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md]: https://github.com/autowarefoundation/autoware-documentation/pull/94/commits/1b9ae53151af38fe3af022ccd4ad9a4de8e02cd6 ?

Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md]: Yeah, thank you! It's clear.

And what do you think about this?

> Also, if you use if available, I feel we should write the case of if not available.

Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md]: hmm makes sense. I recommend them to create an issue for it here.

**Review by kenji-miyake — APPROVED** [review]
Great! :+1:

File: docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md
Comment (xmfcx) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md]: https://github.com/autowarefoundation/autoware-documentation/pull/94/commits/4b524c6998dc36a8658982fc5c24641cb2d926d5 do you think this is ok?

Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md]: Yes, perfect!

File: docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md
Comment (xmfcx) [inline docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:87]: @maxime-clem could you also check here following from https://github.com/autowarefoundation/autoware_msgs/pull/11#discussion_r854781463

Comment (maxime-clem) [reply docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md:87]: Looks good to me, thank you :+1:

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**
